### ISSUE TITLE
The back button does not work correctly in chrome 

### ISSUE THREAD

**Issue by squarfed** [okgrow/router-autoscroll#44]
Running `examples/flow-router-example` and using the back button on chrome the page appears scrolled to the top. Running chrome 51.0.2704.84 on MacOSX.

**Comment by JWPapi** [comment]
Same for me. Is it working on mobile?

**Comment by JWPapi** [comment]
its not just Chrome, its everywhere..

**Comment by JWPapi** [comment]
I have found that https://github.com/tomwasd/flow-router-autoscroll/ has a working back button (at least for Flow Router). tomaswds package is based on this one. So it should be pretty easy to implement his solution into this newer package. 

I don't want to use tomaswds package since it doesnt get updates and some other stuff is not working there(anchor tags) f.e.

**Comment by RemyLespagnol** [comment]
+1

**Comment by SierraGolf** [comment]
I am using iron router and had the same problem. I ended up inlining this package into my project.

The problem is here: https://github.com/okgrow/router-autoscroll/blob/b831cda/client/router-autoscroll.js#L82

At least with iron router (no idea about flow router) this will not scroll when the route is not ready, yet. So for example if the subscriptions takes a while to become ready you might not get the scroll.

I fixed it by changing the used hook from `onRun` to `onBeforeAction` and adding an additional dictionary to keep track if the scroll already happened for that page.

Here my solution in pseudo code:
- on before action
  - if scroll NOT already executed for route
    - scroll
    - save flag that scroll was executed for route
- on stop
  - save scroll position of route
  - reset flag that scroll was executed for route
